source/auth/auth.rst

```diff
@@ -250,6 +250,48 @@ or "SCRAM-SHA-256").
 The cache entry value MUST be either the ``saltedPassword`` parameter or the
 combination of the ``clientKey`` and ``serverKey`` parameters.
 
+Speculative Authentication via Handshake
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:since: 4.4
+
+The ``isMaster`` handshake supports a new argument, ``speculativeSaslStart``,
+provided as a BSON object. Clients specifying this argument to ``isMaster`` will
+speculatively include the first ``saslStart`` command of an authentication handshake.
```

Comment:
If the application does not specify a mechanism (which is very likely), which mechanism should the driver default to in the speculative saslStart?  This seems like a problem because if we pick wrong the effort is wasted.

Reply:
I talked with @sgolemon about this. If a password is provided, the fallback mechanism should be SCRAM-SHA-256. I have added a statement addressing that in this section.